packages/flutter/lib/src/material/bottom_app_bar.dart

```diff
@@ -152,7 +152,11 @@ class _BottomAppBarClipper extends CustomClipper<Path> {
   }
 
   @override
-  bool shouldReclip(covariant _BottomAppBarClipper oldClipper) {
-    return oldClipper.geometry != geometry;
+  bool shouldReclip(CustomClipper<Path> oldClipper) {
+    if (oldClipper.runtimeType != _BottomAppBarClipper)
```

Comment:
this is impossible. oldClipped is guaranteed to have the same runtimeType as `this`
(unless there's a bug in the code that calls `shouldReclip`, which is possible)